commit b1aedc4f44b93aff47ce4bbeff097d8cd52ae51a
Author: Mike Lam <lam@cs.umd.edu>
Date:   Wed Jan 18 12:51:46 2012 -0500

    Added PatchModifier::overwrite

diff --git a/patchAPI/h/PatchModifier.h b/patchAPI/h/PatchModifier.h
index 39747d0..10e6eee 100644
--- a/patchAPI/h/PatchModifier.h
+++ b/patchAPI/h/PatchModifier.h
@@ -81,6 +81,9 @@ class PatchModifier {
                                             bool trust = false, 
                                             Address newlast = (Address)-1);
    
+   // Overwrite every byte of a block's contents with the provided value.
+   PATCHAPI_EXPORT static bool overwrite(PatchBlock *, unsigned char val);
+
    // Remove a block from the CFG; the block must be unreachable
    // (that is, have no in-edges) unless force is true.
    PATCHAPI_EXPORT static bool remove(std::vector<PatchBlock *> &blocks, bool force = false);
diff --git a/patchAPI/src/PatchModifier.C b/patchAPI/src/PatchModifier.C
index 2f1ab9f..258d898 100644
--- a/patchAPI/src/PatchModifier.C
+++ b/patchAPI/src/PatchModifier.C
@@ -113,6 +113,22 @@ PatchBlock *PatchModifier::split(PatchBlock *block, Address addr, bool trust, Ad
 }
 
 
+bool PatchModifier::overwrite(PatchBlock *block, unsigned char val) {
+    ParseAPI::Block *b = block->block();
+    Offset off = b->start();
+    ParseAPI::SymtabCodeRegion *r = 
+        dynamic_cast<ParseAPI::SymtabCodeRegion*>(b->region());
+    if (r == NULL) return false;
+    Offset region_off = (Offset)r->getPtrToInstruction(off) - 
+                        (Offset)r->symRegion()->getPtrToRawData();
+    bool success = false;
+    while (off++ < b->end()) {
+        success = r->symRegion()->patchData(region_off++, (void*)&val, 1);
+        if (!success) return false;
+    }
+    return true;
+}
+
 
 
 InsertedCode::Ptr PatchModifier::insert(PatchObject *obj, void *start, unsigned size, Address base) {
