commit 3253ec969410ad5987593f2444c6a93b98c1134e
Author: Mike Lam <lam@cs.umd.edu>
Date:   Tue May 1 12:34:21 2012 -0400

    Save RAM by not building address mappings

diff --git a/dyninstAPI/src/addressSpace.C b/dyninstAPI/src/addressSpace.C
index fd3cbc3..a65597e 100644
--- a/dyninstAPI/src/addressSpace.C
+++ b/dyninstAPI/src/addressSpace.C
@@ -1816,7 +1816,7 @@ bool AddressSpace::relocateInt(FuncSet::const_iterator begin, FuncSet::const_ite
   }
 
   // Build the address mapping index
-  relocatedCode_.back()->createIndices();
+  //relocatedCode_.back()->createIndices();
     
   // Kevin's stuff
   cm->extractDefensivePads(this);
